src/prisma-service.ts

```diff
@@ -0,0 +1,956 @@
+});
+const fromCreatedAtOrdinalCursor = (row) => ({
+  ...fromCreatedAtCursor(row),
+  ...fromOrdinalCursor(row)
+});
+
+export const globalSnapshots = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return await paginatedResults(
+    event,
+    toCreatedAtOrdinalCursor,
+    fromCreatedAtOrdinalCursor,
+    {
+      include: { dag_blocks: true },
+      orderBy: [
+        { created_at: Prisma.SortOrder.desc },
+        { ordinal: Prisma.SortOrder.desc }
+      ]
+    },
+    prisma.global_snapshots.findMany,
+    globalSnapshotsResponse
+  );
+};
+
+export const globalSnapshot = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { term } = event.pathParameters || {};
+    if (!term) return missingParameterResponse('term');
+
+    const snapshot = await prisma.global_snapshots.findUnique({
+      where: extractTermParam(term),
+      include: { dag_blocks: true }
+    });
+
+    return respond(snapshot, globalSnapshotResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const globalSnapshotRewards = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  const { term } = event.pathParameters || {};
+  if (!term) return missingParameterResponse('term');
+
+  if (!(await global_snapshot_exists(term))) return notFoundResponse();
+
+  const toCursor = (row) => ({
+    global_snapshot_hash_destination_addr: {
+      global_snapshot_hash: row.global_snapshot_hash,
+      destination_addr: row.destination_addr
+    }
+  });
+  const fromCursor = (row) => ({
+    global_snapshot_hash: row.global_snapshot_hash,
+    destination_addr: row.destination_addr
+  });
+
+  return await paginatedResults(
+    event,
+    toCursor,
+    fromCursor,
+    {
+      where: { global_snapshots: { ...extractTermParam(term) } },
+      orderBy: [{ created_at: Prisma.SortOrder.asc }]
+    },
+    prisma.dag_reward_transactions.findMany,
+    rewardsResponse
+  );
+};
+
+export const globalSnapshotTransactions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { term } = event.pathParameters || {};
+    if (!term) return missingParameterResponse('term');
+
+    if (!(await global_snapshot_exists(term))) return notFoundResponse();
+
+    const query = {
+      where: {
+        dag_blocks: { global_snapshots: { ...extractTermParam(term) } }
+      },
+      include: {
+        dag_blocks: {
+          select: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      },
+      orderBy: [{ created_at: Prisma.SortOrder.desc }]
+    };
+
+    return await paginatedResults(
+      event,
+      toCreatedAtOrdinalCursor,
+      fromCreatedAtOrdinalCursor,
+      query,
+      prisma.dag_transactions.findMany,
+      dagTransactionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const block = async (
```

Comment:
Watch the overloading of the `block` variable here. It would be a good idea to have a consistent naming convention for the endpoint functions like `getBlock`.